hledger-lib/Hledger/Utils.hs

```diff
@@ -186,12 +184,7 @@ readFileOrStdinPortably f = openFileOrStdin f ReadMode >>= readHandlePortably
     openFileOrStdin f m   = openFile f m
 
 readHandlePortably :: Handle -> IO Text
-readHandlePortably h = do
-  hSetNewlineMode h universalNewlineMode
-  menc <- hGetEncoding h
-  when (fmap show menc == Just "UTF-8") $  -- XXX no Eq instance, rely on Show
-    hSetEncoding h utf8_bom
-  T.hGetContents h
+readHandlePortably h = hSetNewlineMode h universalNewlineMode *> TIO.hGetContents h
```

Comment:
IIRC, setting utf8_bom encoding made reading utf8 files more robust, by also accepting the utf8 bom prefix that some of them have. Why isn't this needed any more ?

Reply:
The reasoning was that I read something which recommended against it, which I thought was in the `with-utf8` documentation, but I cannot find it anymore. I can leave it in if you'd prefer.

Reply:
I think the issue is that `with-utf8` already sets the encoding, and so if we do it ourselves it will just be overridden. If setting the BOM is indeed desirable (which I have little knowledge of), it should probably be done upstream.